workflow/executor/docker/docker_notwin.go

```diff
@@ -0,0 +1,25 @@
+// +build !windows
```

Comment:
This is not quite the right way to do this. To have different versions of code for different platforms please refer to `chroot_windows.go` vs `chroot_darwin.go`.

Reply:
I implemented it as such because there's no difference between linux & darwin, hence only a difference between windows and not windows is needed. 
As it works quite well like this (and I've seen [codebases](https://github.com/golang/go/blob/master/src/internal/testenv/testenv_notwin.go) using this, also a [blog post references this](https://dave.cheney.net/2013/10/12/how-to-use-conditional-compilation-with-the-go-build-tool)) I'm not sure what you mean with "This is not quite the right way to do this". Possibly you mean that in the argo project you prefer to do it the other way. 

Anyway - I'll follow what @lippertmarkus proposed and solve it using `runtime.GOOS` checks, if that's the preferred way.